### PR TITLE
GH-36421: [Java] Enable Support for reading JSON Datasets

### LINKED ISSUE #36421: [Java] Bindings for JSON Dataset
### Describe the enhancement requested

Enable Support for reading JSON Datasets https://github.com/apache/arrow/pull/33732 on Java side

### Component(s)

Java

### PR DESCRIPTION
**Pull request by davisusanibar**
### Rationale for this change

Enable Support for reading JSON Datasets https://github.com/apache/arrow/pull/33732 on Java side

### What changes are included in this PR?

Support for reading JSON Datasets

### Are these changes tested?

Unit test added

### Are there any user-facing changes?

No
* Closes: #36421

### CODE REVIEW

File: java/dataset/src/test/java/org/apache/arrow/dataset/CsvJsonWriteSupport.java [hunk 1]
```diff
@@ -24,17 +24,18 @@
 import java.net.URISyntaxException;
 import java.util.Random;
 
-public class CsvWriteSupport {
+public class CsvJsonWriteSupport {
```
Comment (kou) [inline java/dataset/src/test/java/org/apache/arrow/dataset/CsvJsonWriteSupport.java:27]: How about using `TextBasedWriteSupport` or something?
If we use this style, we may need to add more formats to class name such as `CsvJsonXXXYYYWriteSupport` in the future.

**Review by lidavidm — APPROVED** [review]
+1 to Kou's comment

File: java/dataset/src/test/java/org/apache/arrow/dataset/TextBasedWriteSupport.java [hunk 2]
```diff
@@ -24,17 +24,18 @@
...
-public class CsvWriteSupport {
+public class TextBasedWriteSupport {
   private final URI uri;
   private final Random random = new Random();
 
-  public CsvWriteSupport(File outputFolder) throws URISyntaxException {
-    uri = new URI("file", outputFolder.getPath() + "/" + "generated-" + random.nextLong() + ".csv", null);
+  public TextBasedWriteSupport(File outputFolder, String jsonOrCsvFileExtension) throws URISyntaxException {
+    uri = new URI("file", outputFolder.getPath() + File.separator +
+        "generated-" + random.nextLong() + jsonOrCsvFileExtension, null);
```
Comment (kou) [inline java/dataset/src/test/java/org/apache/arrow/dataset/TextBasedWriteSupport.java:33]:
```suggestion
  public TextBasedWriteSupport(File outputFolder, String fileExtension) throws URISyntaxException {
    uri = new URI("file", outputFolder.getPath() + File.separator +
        "generated-" + random.nextLong() + fileExtension, null);
```

File: java/dataset/src/test/java/org/apache/arrow/dataset/TextBasedWriteSupport.java [hunk 3]
```diff
@@ -24,17 +24,18 @@
...
   }
 
-  public static CsvWriteSupport writeTempFile(File outputFolder, String... values)
+  public static TextBasedWriteSupport writeTempFile(File outputFolder, String jsonOrCsvFileExtension, String... values)
```
Comment (kou) [inline java/dataset/src/test/java/org/apache/arrow/dataset/TextBasedWriteSupport.java:36]:
```suggestion
  public static TextBasedWriteSupport writeTempFile(File outputFolder, String fileExtension, String... values)
```

File: java/dataset/src/test/java/org/apache/arrow/dataset/TextBasedWriteSupport.java [hunk 4]
```diff
@@ -24,17 +24,18 @@
...
       throws URISyntaxException, IOException {
-    CsvWriteSupport writer = new CsvWriteSupport(outputFolder);
+    TextBasedWriteSupport writer = new TextBasedWriteSupport(outputFolder, jsonOrCsvFileExtension);
```
Comment (kou) [inline java/dataset/src/test/java/org/apache/arrow/dataset/TextBasedWriteSupport.java:38]:
```suggestion
    TextBasedWriteSupport writer = new TextBasedWriteSupport(outputFolder, fileExtension);
```

**Review by lidavidm — APPROVED** [review]
Thanks!

**Review by kou — APPROVED** [review]
+1

**Pull request merged**